### PR TITLE
Particles: More Specializations

### PR DESCRIPTION
**Pull request by ax3l**
Add a few more template specializations as used in ImpactX and HiPACE++ particle containers.

### CODE REVIEW

File: src/Particle/ArrayOfStructs.cpp [hunk 1]
```diff
@@ -80,7 +80,8 @@ void make_ArrayOfStructs(py::module &m)
 }
 
 void init_ArrayOfStructs(py::module& m) {
-    make_ArrayOfStructs<7,0> (m);
+    make_ArrayOfStructs< 0, 0> (m);
```
Comment (ax3l) [inline src/Particle/ArrayOfStructs.cpp:83]: @RTSandberg this one should cause no problems, does it?
```C++
     make_ArrayOfStructs< 0, 0> (m);
```

Comment (RTSandberg) [reply src/Particle/ArrayOfStructs.cpp:83]: I can't think of any reason it should cause problems, we even have some tests in place to keep the user from accidentally referencing non-existent real or int data

File: src/Particle/Particle.cpp [hunk 2]
```diff
@@ -321,8 +321,11 @@ void init_Particle(py::module& m) {
...
-    make_Particle<1, 1> (m);
-    make_Particle<2, 1> (m);
-    make_Particle<3, 2> (m);
+    make_Particle< PIdx::nRealAttribs, PIdx::nIntAttribs > (m);
+    make_Particle< 1, 1 > (m);
+    make_Particle< 2, 1 > (m);
+    make_Particle< 3, 2 > (m);
+    make_Particle< 4, 0 > (m);  // HiPACE++ 22.07
+    make_Particle< 5, 0 > (m);  // ImpactX 22.07
+    make_Particle< 37, 1> (m);  // HiPACE++ 22.07
```
Comment (RTSandberg) [inline src/Particle/Particle.cpp:330]: Thanks for the comments indicating which versions are for which codes

**Review by RTSandberg — COMMENTED** [review]
Looks good to me

File: src/Particle/ArrayOfStructs.cpp (hunk 1 above)
Comment (ax3l) [reply src/Particle/ArrayOfStructs.cpp:83]: Awesome, thanks! :)

File: src/Particle/Particle.cpp (hunk 2 above)
Comment (ax3l) [reply src/Particle/Particle.cpp:330]: Sure thing, thanks :)

Thank you for documenting the task in #49 as well :sparkles:

**Pull request merged**